core-bundle/src/Resources/contao/controllers/BackendMain.php

```diff
@@ -219,7 +219,14 @@ protected function output()
 	{
 		$this->Template->setData($this->compileTemplateData($this->Template->getData()));
 
-		return $this->Template->getResponse();
+		$response = $this->Template->getResponse();
+
+        if (Input::get('popup') !== null)
+        {
+            $response->headers->set('X-Frame-Options', 'sameorigin');
```

Comment:
As discussed in the Contao call, the correct solution would be sending a CSP `frame-ancestors: self` directive instead of the legacy `X-Frame-Options` header.